### ISSUE TITLE
Allow passing multiple env variables and volumes to custom tasks

### ISSUE THREAD

**Issue by anayden** [neuro-inc/neuro-flow#215]
I need to be able to pass arbitrary environment variables and volume specifications to custom tasks defined in `github.com/neuro-actions`. For instance, https://github.com/neuro-actions/remote_debug/blob/master/action.yml might require additional volumes and env vars for user's secrets.

**Comment by asvetlov** [comment]
@romasku have a proposal about envs. Please elaborate here

**Comment by romasku** [comment]
Hello, I think we should try to find some more or less generic solution that allows to pass a variable number of arguments and key-value pairs to an action and then be able to embed them to the YAML list or mapping.

As action inputs are passed as strings, the data should be serialized somehow. As we already have `to_json` and `from_json`, I think we should suggest them for such complex parameters. Because sometimes a simple newline-separated list is acceptable, we should add a `split()` function to convert such string into a sequence inside an action.

Then, we can define some special YAML tags that define merging entries:
``` 
volumes:
  - inline_entry
  - !append ${{ array }}
  - !append ${{ split(newline_separted_str) }}  
env:
  key: value
  key_name_does_not_matter: !extend ${{ mapping }}
```

As this solution is not so easy to implement, here are some less generic but much easier to do variant:

We can just embed newline splitting logic to `volumes` processing code. And for the `env` attribute, we can select some special key that will enable extending logic:
```
env:
  NORMAL_ENV: value
  "**": ${{ mapping }}  # Use mapping keys as env variables names, and values as values.
```
Here `"**"` is the special key for extending `env` mapping (mimics `**` in python).

Personally, I would suggest the first solution, as it will solve this issue in general, without introducing that special cases for some attributes.

**Comment by asvetlov** [comment]
I can propose an alternative solution.

Assume the case is complex enough; the complex case should be solved by embedded expressions.
We need to add expression types that calculate `list` and `dict` correspondingly (all existing expressions keep returning `str`).

By this, the example above can be written as:
```
volumes:  ${{ ['inline_entry'] + array + split(newline_separted_str) }}  # merge lists
env: ${{ {'key': 'value'} | mapping }}  # use | for joining dicts as available in Python 3.9
```

Thoughts?

**Comment by romasku** [comment]
In general, your solution looks prettier as for me, but I cannot fully understand how it will work. 

Our expression can work like template strings (they support the following syntax):
```
${{ value }}_constant_${{ another value }}
```

So, for `volumes` and `env` as in your example, should we completely disallow template-string-like behavior or agree that the following is valid?
```
volumes: '["volume-def-1", "volume-def-2"]'
env: '{VAR1: "value"}'
---
volumes: ${{params.left_square_bracket}}"volume-def-1", "volume-def-2"]
env: ${{ params.param_that_contains_map_serialized_to_json }}
```

Both options are more or less OK for me: disabling such behavior will require a lot of work and will add some special case, and allowing it enables some strange syntax.

**Comment by serhiy-storchaka** [comment]
For lists, what if use different syntax for substitutions?
```
volumes:
  - inline_entry
  - @{{ array }}
  - @{{ split(newline_separted_str) }}
```

**Comment by asvetlov** [comment]
Honestly, I want to not increase the number of syntaxes but reuse the basic syntax when possible.

**Comment by asvetlov** [comment]
Fixed by #236
